README.md

```diff
@@ -1,27 +1,456 @@
+```
+
+| Parameter           | Description                              |
+| :------------------ | :--------------------------------------- |
+| **successCallback** | Executed if the method execution succeed |
+| **errorCallback**   | Executed if the method execution failed  |
+
+#### <span style="color: #6C7E8F">update</span>
+The update operation checks for update to the current card reader and initiates an update if needed. The update can either be a software update or a configuration update.
+
+```javascript 
+cordova.plugins.Handpoint.update({}, successCallback, errorCallback)
+```
+
+| Parameter           | Description                              |
+| :------------------ | :--------------------------------------- |
+| **successCallback** | Executed if the method execution succeed |
+| **errorCallback**   | Executed if the method execution failed  |
+
+#### <span style="color: #6C7E8F">setLogLevel</span>
+Sets the log level for both the card reader and the SDK.
+
+```javascript 
+cordova.plugins.Handpoint.setLogLevel({
+  level: cordova.plugins.Handpoint.LogLevel.Debug,
+  device: device
+}, successCallback, errorCallback)
+```
+
+| Parameter           | Description                                                                                             |
+| :------------------ | :------------------------------------------------------------------------------------------------------ |
+| **config.level**    | Log level. See the list of [supported log levels](#loglevel)                                            |
+| **config.device**   | Device to configure with log level. One of device object returned by [listDevices method](#listDevices) |
+| **successCallback** | Executed if the method execution succeed                                                                |
+| **errorCallback**   | Executed if the method execution failed                                                                 |
+
+#### <span style="color: #6C7E8F">getDeviceLogs</span>
+Fetches the logs from the device and reports them to the **deviceLogsReady** event.
+
+```javascript 
+cordova.plugins.Handpoint.getDeviceLogs({}, successCallback, errorCallback)
+```
+
+| Parameter           | Description                              |
+| :------------------ | :--------------------------------------- |
+| **successCallback** | Executed if the method execution succeed |
+| **errorCallback**   | Executed if the method execution failed  |
+
+## Event Handling
+If you don't want to deal with callbacks and you prefer to turn **Callbacks** into **Promises** you might want to publish the events received in the callback function passed to `cordova.plugins.Handpoint.eventHandler` method to a [publish-subscribe style event system](https://ionicframework.com/docs/api/util/Events/) and then encapsulate the event handling logic in an [SDK wrapper module](https://github.com/handpoint/handpoint-ionic-app/blob/master/src/services/sdk.service.ts).
```

Comment:
A small introduction would be useful.

"As explained before, all the methods of the plugin are asynchronous and return the result of the execution either to a success or an error **callback**: cordova.plugins.Handpoint.methodName([config], successCallback, errorCallback)."